Observability: Model page (#3743)
TobikoData/sqlmesh: docs/cloud/features/observability/model.md

1. Model anomalies
    1. Did the model suddenly take a really long time to run? 
    2. Is the model failing a lot due to audits or schema evolution? 
2. Downstream impacts 
    1. If the model fails to run, using the lineage you can immediately see what other models are affected 
3. The version that introduced errors
    1. The version history of the model is available to view and can be used to see schema changes that may have contributed to any issues 

## Navigate to a model

There are a number of ways you can navigate to a models page. This method shows you how to find your model directly from the Environments page. 

1. Select "Environments" from the left hand menu
2. Click the environment you want to explore from the list. 
    ![Tobiko Cloud environment page](./model/tcloud_environments.png)
3. Navigate to the Models section and click "Explore" to view available models
    ![Tobiko Cloud environment page explore models link](./model/tcloud_environment_explore-models.png)
4. Browse through the model list and select your desired model to access its detailed information
    ![Tobiko Cloud environment models list](./model/tcloud_model_list.png)

## Model page information

> Each model page presents a comprehensive summary that includes several key components and metrics for monitoring and analysis. From here you can understand or view any 

Reviewer: ```suggestion
Each model page presents a comprehensive summary that includes the key components and metrics used to monitor model behavior. 
```

> anomalies in the run time of the model, and get a sense for the overall processing time and how that is changing (or not!) over time. You can also check other model-critical metrics, 

Reviewer: ```suggestion

From here, you can identify anomalies in the model's run time based on historical run times and how they have been changing over time (or not!).
```

> like the source code for the model, its lineage in relation to other models, the number of versions of this model available to you and even an approximation of how much this model 

Reviewer: ```suggestion

You can also check other critical information, like the model's source code, its lineage relative to other models, its contents in previous versions, and even an approximation of how much it costs (if you have [cost savings set up](../costs_savings.md)).
```

> is costing (if you have [cost savings set up](../costs_savings.md)). 

Reviewer: ```suggestion
```

The following detailed information outlines the different sections:

![Tobiko Cloud model status and metadata](./model/tcloud_model_status-metadata.png)

> - Current status graphs: Provides visual representations of model health through freshness indicators and detailed daily execution graphs

Reviewer: ```suggestion
- Current status graphs: Provide visual representations of model health through freshness indicators and detailed daily execution graphs
```

    - Freshness indicator: Shows the current status of the model and the percentage of up-to-date models in production (as long as this is green, you have nothing to worry about in your production environment)
    - Historical Freshness graph: Gives an at-a-glance picture of the history of the model's freshness.

>         - Green means it's up to date and has run smoothly for every cron interval

Reviewer: ```suggestion
        - Green means it's up to date and has run smoothly for every interval
```

>         - Orange means that it is pending and will run on the latest cron

Reviewer: ```suggestion
        - Orange means that one interval is pending and will be processed on the next run
```

>         - Red means the model is behind by at least one cron interval.

Reviewer: ```suggestion
        - Red means the model has more than one interval waiting to be processed because an interval was not processed during a previous run
```

>     - Daily executions: tells you the length of time it took the model to run on each day displayed. This is a great place to quickly understand any anomalies in run time for the model (both running too long *or* too short)

Reviewer: ```suggestion
    - Daily executions: tells you the length of time it took the model to run on each day. This is a great place to quickly identify anomalies in the model's run time (both running too long *or* too short).
```

> - Model details: Features comprehensive tabs that display summary statistics, complete source code documentation, and interactive model lineage visualizations

Reviewer: ```suggestion
- Model details: Features tabs that display summary statistics, model source code, and interactive model lineage visualizations
```

![Tobiko Cloud model version history](./model/tcloud_model_2.png)

- Version history: Delivers a comprehensive chronological view of all model versions, with detailed information including:
    - Precise timestamp of version promotion
    - Clear indication of change impact (breaking or non-breaking modifications)
    - Direct access to the complete implementation plan code
- Data Warehouse costs: estimates the cost of the model as set up by your team in [cost savings](../costs_savings.md)

![Tobiko Cloud model version history](./model/tcloud_model_3.png)

- Loaded intervals: these periods represent the time spans processed during each job execution, which generally consist of the time between one job and the next. These intervals are crucial for understanding the boundaries of data processing cycles, which may correspond to the start of anomalous model behavior.
    - The table displays the specific model version in effect during that job execution, enabling precise tracking of version-specific outputs
    - Helps track forward-only model changes by maintaining a clear chronological record of modifications, ensuring data consistency and preventing retroactive alterations

>     - provides comprehensive visibility into completed data processing operations, allowing users to monitor progress and verify successful execution of scheduled tasks

Reviewer: ```suggestion
```

    - Precise timestamp of version promotion
    - Clear indication of change impact (breaking or non-breaking modifications)
    - Direct access to the complete implementation plan code
- Data Warehouse costs: estimates the cost of the model as set up by your team in [cost savings](../costs_savings.md)

![Tobiko Cloud model version history](./model/tcloud_model_3.png)

> - Loaded intervals: these periods represent the time spans between consecutive cron job executions, from the start of one cron job to the end of the next cron job. These intervals are crucial for understanding the boundaries of data processing cycles

Reviewer: ```suggestion
- Loaded intervals: these periods represent the time spans processed during each job execution, which generally consist of the time between one job and the next. These intervals are crucial for understanding the boundaries of data processing cycles, which may correspond to the start of anomalous model behavior.
```

>     - the table displays which specific model version was responsible for generating and processing data during each distinct cron interval, enabling precise tracking of version-specific outputs

Reviewer: ```suggestion
    - The table displays the specific model version in effect during that job execution, enabling precise tracking of version-specific outputs
```

>     - helps track forward-only model changes by maintaining a clear chronological record of modifications, ensuring data consistency and preventing retroactive alterations

Reviewer: ```suggestion
    - Helps track forward-only model changes by maintaining a clear chronological record of modifications, ensuring data consistency and preventing retroactive alterations
```
